lib/Modules/ADHooks/ADHooks.psm1

```diff
@@ -1458,3 +1550,52 @@ function Invoke-ADSPNRelationChangedHook {
         }
     }
 }
+
+function Invoke-UpdateStatusHook {
+    if(!(Confirm-Leader)){
+        Write-JujuWarning "Unit is not leader. Skipping the rest of the hook"
+        return
+    } else {
```

Comment:
No need to have the code in the `else` block. Just leave it after the `if``, as we return early in case the `then` branch is executed.